src/java.base/macosx/classes/sun/nio/ch/KQueue.java

```diff
@@ -25,17 +25,28 @@
 
 package sun.nio.ch;
 
-import java.io.IOException;
-import jdk.internal.misc.Unsafe;
+import jdk.internal.ffi.generated.errno.errno_h;
+import jdk.internal.ffi.generated.timespec.timespec;
+import jdk.internal.ffi.generated.kqueue.kevent;
+import jdk.internal.ffi.generated.kqueue.kqueue_h;
+import jdk.internal.ffi.util.FFMUtils;
+import jdk.internal.foreign.*;
+
+import java.io.*;
+import java.lang.foreign.*;
+
+import static java.lang.foreign.MemorySegment.NULL;
 
 /**
  * Provides access to the BSD kqueue facility.
  */
 
-class KQueue {
-    private KQueue() { }
+final class KQueue {
 
-    private static final Unsafe unsafe = Unsafe.getUnsafe();
+    private static final int KQUEUE_ERROR_VALUE = -1;
+    private static final BufferStack POOL = BufferStack.of(timespec.layout());
```

Comment:
Perhaps we could use `Math::max` to determine at runtime how much memory to allocate in the BufferStack?